library/std/src/io/stdio.rs

```diff
@@ -670,7 +663,7 @@ impl Write for Stdout {
     }
     #[inline]
     fn is_write_vectored(&self) -> bool {
-        io::Write::is_write_vectored(&&*self)
+        stdout_raw().is_write_vectored()
```

Comment:
Since `LineWriter::is_write_vectored` [forwards to the inner reader](https://github.com/rust-lang/rust/blob/dd01122b5c62a04e64b4109c5576eeea9ae4145b/library/std/src/io/buffered/linewriter.rs#L201-L203), I think it is better to avoid that indirection, as this allows the function to be inlined (it returns a constant for all current systems, so all `if Stdout.is_write_vectored() { ... }` can be optimized out).

Reply:
That's true today, but it might not be in the future. If adding an `#[inline]` to LineWriter helps, I'd be fine with that, though it probably would be best in a separate PR, just like this change.